src/Integration.Vsix/CFamily/ProcessRunner.cs

```diff
@@ -116,66 +120,87 @@ public bool Execute(ProcessRunnerArguments runnerArgs)
-
-            if (succeeded)
-            {
-                process.WaitForExit(); // Give any asynchronous events the chance to complete
-            }
-
-            // false means we asked the process to stop but it didn't.
-            // true: we might still have timed out, but the process ended when we asked it to
-            if (succeeded)
+            try
             {
-                LogDebug(CFamilyStrings.MSG_ExecutionExitCode, process.ExitCode);
-                ExitCode = process.ExitCode;
+                var succeeded = process.WaitForExit(runnerArgs.TimeoutInMilliseconds);
 
-                if (process.ExitCode != 0 && !runnerArgs.CancellationToken.IsCancellationRequested)
+                if (succeeded)
                 {
-                    LogError(CFamilyStrings.ERROR_ProcessRunner_Failed, process.ExitCode);
+                    process.WaitForExit(); // Give any asynchronous events the chance to complete
                 }
-            }
-            else
-            {
-                ExitCode = ErrorCode;
 
-                try
+                // false means we asked the process to stop but it didn't.
+                // true: we might still have timed out, but the process ended when we asked it to
+                if (succeeded)
                 {
-                    process.Kill();
-                    LogWarning(CFamilyStrings.WARN_ExecutionTimedOutKilled, runnerArgs.TimeoutInMilliseconds,
-                        runnerArgs.ExeName);
+                    LogDebug(CFamilyStrings.MSG_ExecutionExitCode, process.ExitCode);
+                    ExitCode = process.ExitCode;
+
+                    if (process.ExitCode != 0 && !runnerArgs.CancellationToken.IsCancellationRequested)
+                    {
+                        LogError(CFamilyStrings.ERROR_ProcessRunner_Failed, process.ExitCode);
+                    }
                 }
-                catch
+                else
                 {
-                    LogWarning(CFamilyStrings.WARN_ExecutionTimedOutNotKilled, runnerArgs.TimeoutInMilliseconds,
-                        runnerArgs.ExeName);
+                    ExitCode = ErrorCode;
+
+                    try
+                    {
+                        process.Kill();
+                        LogWarning(CFamilyStrings.WARN_ExecutionTimedOutKilled, runnerArgs.TimeoutInMilliseconds,
+                            runnerArgs.ExeName);
+                    }
+                    catch
+                    {
+                        LogWarning(CFamilyStrings.WARN_ExecutionTimedOutNotKilled, runnerArgs.TimeoutInMilliseconds,
+                            runnerArgs.ExeName);
+                    }
                 }
-            }
 
-            return succeeded && ExitCode == 0;
+                return succeeded && ExitCode == 0;
+            }
+            catch (InvalidOperationException ex) when (ex.Message == "No process is associated with this object.")
```

Comment:
Since it runs on a different task, when we cancel the process, an exception is thrown since the process is disposed (but it's not ObjectDisposedException... it's InvalidOperationException with that message).

Reply:
Exception messages are localized so checking against the en-us version isn't going to work reliably.

Reply:
Also, does this mean we aren't reliably writing the success/failure/timeout status to the output window?